src/flake8_requirements/checker.py

```diff
@@ -656,13 +656,21 @@ def get_mods_3rd_party(self):
         mods_3rd_party = ModuleSet()
         # Get 3rd party module names based on requirements.
         for requirement in self.get_mods_3rd_party_requirements():
-            modules = [project2module(requirement.project_name)]
-            if modules[0] in self.known_3rd_parties:
-                modules = self.known_3rd_parties[modules[0]]
-            elif modules[0] in self.known_host_3rd_parties:
-                modules = self.known_host_3rd_parties[modules[0]]
-            elif modules[0] in self.known_modules:
-                modules = self.known_modules[modules[0]]
+            mod = project2module(requirement.project_name)
+            modules = (
```

Comment:
What is the exact usage for that? I mean, why you are collecting modules from all sets (known 3rd parties, host 3rd parties and modules specified by user)? You are right that something is wrong here, but I would simply change the order of "ifs". So, the user specified modules will be check first. This will allow to overwrite build-ins with user defined modules.

Reply:
My thinking was that the user defined modules can be used to supplement the library specified ones, rather than completely overriding them. It would mean that you don't have to list every module, just the ones missing from known 3rd parties, as ling as the modules in known 3rd parties are actually correct. Not entirely sure what known_host_3rd_parties is, but I figured just taking the union of these sets would be fine, and it means I didn't have to reason too much about ordering.

Do you have a concrete example for when doing it this way would lead to wrong/unintended results? I guess, if there are false positives in one of the three collections, that could mask problems for the user. But it is hard to say which approach is better, also in that case.

Reply:
> My thinking was that the user defined modules can be used to supplement the library specified ones, rather than completely overriding them.

Ok, I see your point. But my concerns are exactly about that. In most cases user configuration overrides build-in config entirely. So, such approach is the most flexible. I know that it might be inconvenient in some cases, but it shouldn't be here. The list of modules is typically short, and since user is overwriting package, they can add all required modules.

So, please reorder "ifs" and that should be it. Otherwise, there would have to be an extra info in the help message, that user config is meged with build-in one.

Reply:
Done.